React/Views/RCTWebView.m

```diff
@@ -248,6 +270,13 @@ - (void)webView:(__unused UIWebView *)webView didFailLoadWithError:(NSError *)er
 
 - (void)webViewDidFinishLoad:(UIWebView *)webView
 {
+  if (_messagingEnabled) {
+    NSString *source = [NSString stringWithFormat:
+      @"window.postMessage = function(message) {"
+      "  window.location = '%@://%@?' + encodeURIComponent(JSON.stringify(message));"
+      "};", RCTJSNavigationScheme, RCTJSPostMessageHost];
```

Comment:
@jacobp100 With this code any shims will continue to work, as well as the `onMessage` prop. The current behaviour is hard to predict and will need documentation since it will overwrite any shims.